pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -100,6 +101,129 @@ func (c *Controller) GetAppliedToGroupNum() int {
 	return c.ruleCache.GetAppliedToGroupNum()
 }
 
+func ipNetConverter(ipNets ...v1beta1.IPNet) (ret []networking.IPNet) {
+	for _, ipNet := range ipNets {
+		ret = append(ret, networking.IPNet{
+			IP:           networking.IPAddress(ipNet.IP),
+			PrefixLength: ipNet.PrefixLength,
+		})
+	}
+	return ret
+}
+
+func ipBlocksConverter(blocks []v1beta1.IPBlock) (ret []networking.IPBlock) {
+	for _, block := range blocks {
+		ret = append(ret, networking.IPBlock{
+			CIDR:   ipNetConverter(block.CIDR)[0],
+			Except: ipNetConverter(block.Except...),
+		})
+	}
+	return ret
+}
+
+func portsConverter(ports ...v1beta1.NamedPort) (ret []networking.NamedPort) {
+	for _, port := range ports {
+		ret = append(ret, networking.NamedPort{
+			Port:     port.Port,
+			Name:     port.Name,
+			Protocol: networking.Protocol(port.Protocol),
+		})
+	}
+	return ret
+}
+
+func serviceConverter(services ...v1beta1.Service) (ret []networking.Service) {
+	for _, service := range services {
+		protocol := networking.Protocol(*service.Protocol)
+		ret = append(ret, networking.Service{
+			Protocol: &protocol,
+			Port:     service.Port,
+		})
+	}
+	return ret
+}
+
+func (c *Controller) GetNetworkPolicies() (ret []networking.NetworkPolicy) {
+	c.ruleCache.policySetLock.RLock()
+	defer c.ruleCache.policySetLock.RUnlock()
+	for uid := range c.ruleCache.policySet {
+		np := networking.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{Name: uid},
+		}
+		rules, _ := c.ruleCache.rules.ByIndex(policyIndex, uid)
+		for _, ruleObj := range rules {
+			rule := ruleObj.(*rule)
+			np.Rules = append(np.Rules, networking.NetworkPolicyRule{
+				Direction: networking.Direction(rule.Direction),
+				From: networking.NetworkPolicyPeer{
+					AddressGroups: rule.From.AddressGroups,
+					IPBlocks:      ipBlocksConverter(rule.From.IPBlocks),
+				},
+				To: networking.NetworkPolicyPeer{
+					AddressGroups: rule.To.AddressGroups,
+					IPBlocks:      ipBlocksConverter(rule.To.IPBlocks),
+				},
+				Services: serviceConverter(rule.Services...),
+			})
+			np.AppliedToGroups = append(np.AppliedToGroups, rule.AppliedToGroups...)
+		}
+		ret = append(ret, np)
+	}
+	return ret
```

Comment:
I didn't know you are constructing the policy back from stored items, from perspective of debugging, we care rules only, I think it's ok to just dump rules.

Reply:
Outdated.